Sources/TuistCore/Graph/Nodes/TargetNode.swift

```diff
@@ -113,3 +114,15 @@ public class TargetNode: GraphNode {
         sdkDependencies.contains(where: { $0.name == "XCTest" }) || target.product.testsBundle
     }
 }
+
+private extension PrecompiledNode {
+    var nodeDependencies: [PrecompiledNode] {
+        dependencies.map { $0.node }
+    }
+}
+
+private func recursiveNodeDependencies(_ node: PrecompiledNode) -> [PrecompiledNode] {
```

Comment:
We have a method in `Graph` for traversing the graph making sure there's only one pass through each of the nodes.

Reply:
Oh, didn't see it 👍 Assume we are talking about `Graph.findAll` method, right?